### ISSUE TITLE
Inserting POJO using mongo-reactive doesn't set the id

### ISSUE THREAD

**Issue by ilopmar** [micronaut-projects/micronaut-core#703]
According to [quick-start-pojo](http://mongodb.github.io/mongo-java-driver/3.8/driver/getting-started/quick-start-pojo/) and [bson-pojos](http://mongodb.github.io/mongo-java-driver/3.8/bson/pojos/) to automatically convert mongo documents to POJOs we need to:
- Register a PojoCodecProvider. Done here: https://github.com/micronaut-projects/micronaut-core/blob/e97b8e0759231765ceb07d75eb100edce5fbf51a/configurations/mongo-reactive/src/main/java/io/micronaut/configuration/mongo/reactive/AbstractReactiveMongoConfiguration.java#L148-L152
- Have an empty constructor in the POJO class

This test https://github.com/micronaut-projects/micronaut-core/commit/0eb8a40ad195d8c63823e658779553f6ff1c0cff in the brach `mongo-reactive-insert-pojo`  fails because the `id` is not set after saving the object in mongo. I can see that the `_id` field is generated in mongo but it is always `null` in the instance. 

I'm not really sure if we need to configure anything else on Micronaut or maybe I'm missing something in my example.

I've also tried defining the `id` property as `String` and also annotate it with `@BsonId` but the result is the same.

### Environment Information

- **Operating System**: Linux
- **Micronaut Version:** Micronaut.1.0.0.RC1
- **JDK Version:** openjdk version "1.8.0_181"

**Comment by ilopmar** [comment]
I'm not sure, maybe the user needs to do it manually, I'm not sure. 

Btw @goeh how do you retrieve only the `_id` back in the json? If manually set the id, store the object and render the `user` instance I get something like:
```
{"id":{"timestamp":1539001948,"machineIdentifier":13828076,"processIdentifier":11510,"counter":9463323,"timeSecond":1539001948,"time":1539001948000,"date":1539001948000},"name":"Iván","age":38}
```

Instead of 
```
{"id":"xxxxxxxxxxxxxxx","name":"Iván","age":378}
```

Because jackson is converting the whole `ObjectId` object instead of executing `toString()` on it.

**Comment by graemerocher** [comment]
IMO nothing we can do about this. Micronaut shouldn't get involved in settings ids and this is down to how the driver works

**Comment by goeh** [comment]
I declare id as ObjectId in the POJO, then pojo.setId(new ObjectId()) before insert.

**Comment by ilopmar** [comment]
@goeh Yes, `pojo.getId()` returns the `_id` but rendering the pojo doesn't.

**Comment by ilopmar** [comment]
@graemerocher maybe we can add some kind of support like Spring Data Mongo so the users don't need to manually deal with the `_id`. Maybe annotate the id field with `@MongoId` or similar and let Micronaut to set the id automatically.

**Comment by graemerocher** [comment]
Maybe, but not for 1.0

**Comment by coderish** [comment]
Hi Guys,

 i'v been trying to save a document in mongo as a poc but micronaut doesnt write. However i am able to retrieve the data from mongo. After a long research i was able to see this discussion forum. Can anybody please suggest on how can i resolve my problem?

  public Single save(@Valid @Body UserEntity userEntity) {
        System.out.println("Inside Save method - "+ userEntity.get_id()+": "+ userEntity.getUsername() + userEntity.getPassword() + userEntity.getToken());
        return Single.fromPublisher(
            getCollection().insertOne(userEntity))
                .map(success -> userEntity);
    }


    @BsonId
    private ObjectId _id;
    private String username;
    private String password;
    private String token;

    @BsonCreator
    @JsonCreator
    public UserEntity(
        @JsonProperty("_id") @BsonProperty("_id") ObjectId _id,
        @JsonProperty("username") @BsonProperty("username") String username,
        @JsonProperty("password") @BsonProperty("password") String password,
        @JsonProperty("token") @BsonProperty("token") String token) {
        this.username = username;
        this.password = password;
        this.token = token;
        this._id = _id;
    }

    public UserEntity() {}

getters/setters below

**Comment by hberrayana** [comment]
I managed to get it working and using this in production. I declared a class like this:
```
@BsonDiscriminator
public class SessionData {


    @BsonId
    private String instanceId;


    @BsonProperty(value = "created")
    private Instant created;

    @BsonProperty(value = "updated")
    private Instant updated;

    @BsonProperty(value = "accessed")
    private Instant lastAccessed;


    @BsonProperty(value = "userAgent")
    private String userAgent;

    @BsonProperty(value = "application")
    private String  application;

    @BsonProperty(value = "applicationId")
    private BsonObjectId  applicationId;
}
```

I had to register the package of this class similar to what is written here #840

**Comment by ErikDevlop** [comment]
I'm having he same problem. It does however work perfectly when running on my localhost with mongod version  3.4.15 and Java Driver 3.8.0. But when I switch the connection string to use the MongoDB Atlas with version 4.0.6 then it doesn't return the id after inserting it. It does however create the id in the collection on MongoDB Atlas. Would really like a explaination to this, and a fix :)

**Comment by yhs97** [comment]
Has anyone found a solution to this? I'm facing the same issue- on rendering the POJO object, the _id field is null.

**Comment by cknmca** [comment]
> Has anyone found a solution to this? I'm facing the same issue- on rendering the POJO object, the _id field is null.

Yes.
use the following and add the package eclispe shows in intellisense while insert this code.

CodecRegistry pojoCodecRegistry = fromProviders(PojoCodecProvider.builder().register("com.sample.user").build());
			CodecRegistry codecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),pojoCodecRegistry);

			MongoClientSettings clientSettings = MongoClientSettings.builder().applyConnectionString("mongodb://localhost:27017").codecRegistry(codecRegistry).build();

			try (MongoClient mongoClient = MongoClients.create(clientSettings)) {
				MongoDatabase db = mongoClient.getDatabase("test");
				MongoCollection<UserEntity> collection = db.getCollection(employee, UserEntity.class);


after you can use it document conversion method to add your phone and update it....here test is the db name and employee is the collection create in mongo then use it
